File: app/src/main/java/com/test/kabak/openweather/core/network/TeleportApi.java
package com.test.kabak.openweather.core.network;

import io.reactivex.Single;
import retrofit2.http.GET;
import retrofit2.http.Query;

public interface TeleportApi {
    @GET("cities/")
    Single<SearchCityResponse> getCities(@Query("search") String searchString);

    @GET("cities/")
    Single<Object> getCities(@Query("geonameid") int cityId);
}

File: app/src/main/java/com/test/kabak/openweather/core/network/TeleportApi.kt
package com.test.kabak.openweather.core.network

import com.test.kabak.openweather.core.network.dataClasses.SearchCityResponse

import io.reactivex.Single
import retrofit2.http.GET
import retrofit2.http.Query

interface TeleportApi {
    @GET("cities/")
    fun getCities(@Query("search") searchString: String): Single<SearchCityResponse>
}
